### ISSUE TITLE
Sign in with Apple relation to Auth and API

### ISSUE THREAD

**Issue by amuresia** [aws-amplify/aws-sdk-ios#2520]
**State your question**
I need to integrate Sign In with Apple into my ios app but secure access to GraphQL models
**Which AWS Services are you utilizing?**
Cognito User Pools, Cognito Identity Pools, AppSync
**Provide code snippets (if applicable)**
All the resources are provisioned with the amplify cli.
I have successfully integrated sign in with apple into my app and I can use federated identities like below:
```
AWSMobileClient.default().federatedSignIn(providerName: "appleid.apple.com", token: token) { userState, error in
            if let error = error {
                print("Error in federatedSignIn: \(error)")
                return
            }
            guard let userState = userState else {
                print("userState unexpectedly nil")
                return
            }
            print("federatedSignIn successful: \(userState)")
```
This works and my user is signed in.
The issue is that, for the user to use the GraphQL API, I need to have the following
```
@model
@auth(rules: [
    { allow: owner, operations: [create, update, delete] },
    { allow: private, provider: iam }
])
``` 
This works, however, it gives "God mode" to everybody who logs in via SIWA. I don't want that. I want users to only be able to create, update and delete, entries that they create in the models.
Ideally, I would like to create the user in Cognito User Pools when they use Sign In with Apple, but I am not too sure how to use the drop-in UI because my app is written in swiftUI where there's no out of the box 'navigation controller' that I can provide to the hostedUI show function.

To summarise, I would like guidance for the following points:
1. How do I allow users that sign in with Apple to only create, update and delete entries created by them.
2. When somebody Signs in with Apple, how do I create the user in Cognito User Pools, I am happy to introduce the drop-in UI (at least for now) but could I please get some guidance on how to make it work in SwiftUI

**Comment by palpatim** [comment]
> When somebody Signs in with Apple, how do I create the user in Cognito User Pools, I am happy to introduce the drop-in UI (at least for now) but could I please get some guidance on how to make it work in SwiftUI

`federatedSignIn` only federates into your Cognito IdentityPool. To create an entry in your UserPool, you'll need to set up SIWA as an identity provider for your UserPool, and have your users sign in with HostedUI. You can see more information in our docs https://docs.amplify.aws/lib/auth/hosted-ui/q/platform/ios#automated-setup-with-cli. Unfortunately we don't yet have a public example of using this with SwiftUI, but I'll check on the team to see if anybody's got a work-in-progress they can use to help with this.

Regarding elevated privileges for users federating into your IdentityPool via SIWA: I don't have any particular insight into this. Do you support other authentication providers in your IdentityPool, and do users who sign in with those providers behave as expected? If so, please ensure that the role (either "use default", or the role derived by rules) for SIWA matches the selection rules for the other providers.

If the role assignment for all of your federation providers is the same, and the behavior differs only based on the authentication provider the user used to federate into the pool, please let us know--we may need to investigate this on the service side to see if there are some subtleties in the setup.

**Comment by palpatim** [comment]
Feel free to reopen if you have more questions.
